### PR TITLE
feat(console): add experience form fields

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement form fields in tab "Experience".

In order to make PR more clear, form field components will be implemented in other PRs, you can see those TODOs in code comment.

## Linear Issue Reference

LOG-1729

## Testing

image

### CODE REVIEW

**Review by simeng-li — APPROVED** [review]
LGTM

**Comment by demonzoo** [comment]
Generally it looks different from the UX design. Am I looking at the out-dated design pages again?
image

**Comment by wangsijie** [comment]
@demonzoo because many components are under constrution?

**Comment by IceHe** [comment]
Should be `Logo` & `Slogan`

image

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      <FormField isRequired title="admin_console.sign_in_exp.branding.ui_style">
+        {/* TODO: LOG-2153 plain radio */}
+        <Controller
+          name="branding.style"
+          control={control}
+          defaultValue={BrandingStyle.Logo_Slogan}
+          render={({ field: { onChange, value, name } }) => (
+            <RadioGroup value={value} name={name} onChange={onChange}>
+              <Radio value={BrandingStyle.Logo_Slogan}>
+                {t('sign_in_exp.branding.app_logo')} + {t('sign_in_exp.branding.text')}
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx:45]: Should be `Logo` & `Slogan`

image

**Review by IceHe — CHANGES_REQUESTED** [review]

File: packages/phrases/src/locales/en.ts [hunk 2]
```diff
@@ -324,6 +324,7 @@ const translation = {
       branding: {
         title: 'BRANDING',
         primary_color: 'Primary color',
+        dark_primary_color: 'Primary color(Dark)',
```
Comment (IceHe) [inline packages/phrases/src/locales/en.ts:327]: Should be `'Primary color (Dark)'`

image

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      <FormField isRequired title="admin_console.sign_in_exp.branding.slogan">
+        <TextInput {...register('branding.slogan', { required: true })} />
+      </FormField>
+      <div className={styles.title}>{t('sign_in_exp.terms_of_use.title')}</div>
+      <FormField isRequired title="admin_console.sign_in_exp.terms_of_use.enable">
+        <TextInput {...register('termsOfUse.enabled', { required: true })} />
+        <div>{t('sign_in_exp.terms_of_use.description')}</div>
+      </FormField>
+      <FormField isRequired title="admin_console.sign_in_exp.terms_of_use.terms_of_use">
+        <TextInput {...register('termsOfUse.contentUrl', { required: true })} />
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx:64]: If Terms-of-use `enabled` is false, `contentUrl` is not required.

Should it be always required here?

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+              <Radio value={BrandingStyle.Logo}>{t('sign_in_exp.branding.app_logo')}</Radio>
+            </RadioGroup>
+          )}
+        />
+      </FormField>
+      <FormField isRequired title="admin_console.sign_in_exp.branding.logo_image_url">
+        <TextInput {...register('branding.logoUrl', { required: true })} />
+      </FormField>
+      <FormField isRequired title="admin_console.sign_in_exp.branding.slogan">
+        <TextInput {...register('branding.slogan', { required: true })} />
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx:56]: If the UI style does not contain a slogan, it's not required.

Should it be always required here?

File: packages/phrases/src/locales/zh-cn.ts [hunk 5]
```diff
@@ -321,6 +321,7 @@ const translation = {
       branding: {
         title: 'BRANDING',
         primary_color: 'Primary color',
+        dark_primary_color: 'Primary color(Dark)',
```
Comment (IceHe) [inline packages/phrases/src/locales/zh-cn.ts:324]: ditto

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx]: updated

File: packages/phrases/src/locales/en.ts (hunk 2 above)
Comment (wangsijie) [reply packages/phrases/src/locales/en.ts]: updated

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx]: updated

File: packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/ExperienceForm.tsx]: updated

File: packages/phrases/src/locales/zh-cn.ts (hunk 5 above)
Comment (wangsijie) [reply packages/phrases/src/locales/zh-cn.ts]: updated

File: packages/console/src/pages/SignInExperience/components/BrandingForm.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      <FormField isRequired title="admin_console.sign_in_exp.branding.ui_style">
+        {/* TODO: LOG-2153 plain radio */}
+        <Controller
+          name="branding.style"
+          control={control}
+          defaultValue={BrandingStyle.Logo_Slogan}
+          render={({ field: { onChange, value, name } }) => (
+            <RadioGroup value={value} name={name} onChange={onChange}>
+              <Radio value={BrandingStyle.Logo_Slogan}>
+                {t('sign_in_exp.branding.logo')} + {t('sign_in_exp.branding.text')}
```
Comment (IceHe) [inline packages/console/src/pages/SignInExperience/components/BrandingForm.tsx:53]: should be `'sign_in_exp.branding.slogan'`

File: packages/phrases/src/locales/en.ts [hunk 7]
```diff
@@ -324,11 +324,12 @@ const translation = {
...
         primary_color: 'Primary color',
-        dark_mode: 'Enable dark mode',
+        dark_primary_color: 'Primary color (Dark)',
+        dark_mode: 'Enable Dark Mode',
         dark_mode_description:
           'Enabling this setting will auto generate the dark mode color. You app won’t have dark mode if it’s turned off.',
         ui_style: 'Define your UI style',
-        app_logo: 'App Logo',
+        logo: 'Logo',
         text: 'Text',
```
Comment (IceHe) [inline packages/phrases/src/locales/en.ts:333]: should be `slogan`

File: packages/phrases/src/locales/zh-cn.ts [hunk 8]
```diff
@@ -321,11 +321,12 @@ const translation = {
...
         primary_color: 'Primary color',
-        dark_mode: 'Enable dark mode',
+        dark_primary_color: 'Primary color (Dark)',
+        dark_mode: 'Enable Dark Mode',
         dark_mode_description:
           'Enabling this setting will auto generate the dark mode color. You app won’t have dark mode if it’s turned off.',
         ui_style: 'Define your UI style',
-        app_logo: 'App Logo',
+        logo: 'Logo',
         text: 'Text',
```
Comment (IceHe) [inline packages/phrases/src/locales/zh-cn.ts:330]: ditto

File: packages/console/src/pages/SignInExperience/components/BrandingForm.tsx (hunk 6 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/BrandingForm.tsx]: updated

File: packages/phrases/src/locales/en.ts (hunk 7 above)
Comment (wangsijie) [reply packages/phrases/src/locales/en.ts]: updated

File: packages/phrases/src/locales/zh-cn.ts (hunk 8 above)
Comment (wangsijie) [reply packages/phrases/src/locales/zh-cn.ts]: updated

**Review by IceHe — APPROVED** [review]
LGTM

**Pull request merged**